src/go/agwd/config/config.go

```diff
@@ -156,7 +188,15 @@ func (c *ConfigManager) Merge(update *config.AgwD) {
 		panic("clone of defaultConfig not *config.AgwD")
 	}
 	proto.Merge(config, update)
-	c.config = config
+	*c.config = *config
+}
+
+// Replace overwrites the managed config.
+func (c *ConfigManager) Replace(replace *config.AgwD) {
+	c.Lock()
+	defer c.Unlock()
+
+	*c.config = *replace
```

Comment:
you probably want to use https://pkg.go.dev/github.com/golang/protobuf/proto#Clone, which creates a deep copy.

(`*x = *y` does a shallow copy)

Reply:
cool just added